include/mc_tasks/ImpedanceTask.h

```diff
@@ -0,0 +1,359 @@
+  void impedanceOrientation(const Eigen::Vector3d & impMOri,
+                            const Eigen::Vector3d & impDOri,
+                            const Eigen::Vector3d & impKOri)
+  {
+    impM_.moment() = mc_filter::utils::clampAndWarn(impMOri, impMOriLower_, impMOriUpper_, "impedanceOrientationM");
+    impD_.moment() = mc_filter::utils::clampAndWarn(impDOri, impDOriLower_, impDOriUpper_, "impedanceOrientationD");
+    impK_.moment() = mc_filter::utils::clampAndWarn(impKOri, impKOriLower_, impKOriUpper_, "impedanceOrientationK");
+  }
+
+  /*! \brief Get the impedance mass parameter represented in the surface. */
+  const sva::ForceVecd & impedanceM() const noexcept
+  {
+    return impM_;
+  }
+
+  /*! \brief Set the impedance mass parameter represented in the surface. */
+  void impedanceM(const sva::ForceVecd & impM)
+  {
+    impM_.force() = mc_filter::utils::clampAndWarn(impM.force(), impMPosLower_, impMPosUpper_, "impedancePositionM");
+    impM_.moment() =
+        mc_filter::utils::clampAndWarn(impM.moment(), impMOriLower_, impMOriUpper_, "impedanceOrientationM");
+  }
+
+  /*! \brief Get the impedance damper parameter represented in the surface. */
+  const sva::ForceVecd & impedanceD() const noexcept
+  {
+    return impD_;
+  }
+
+  /*! \brief Set the impedance damper parameter represented in the surface. */
+  void impedanceD(const sva::ForceVecd & impD)
+  {
+    impD_.force() = mc_filter::utils::clampAndWarn(impD.force(), impDPosLower_, impDPosUpper_, "impedancePositionD");
+    impD_.moment() =
+        mc_filter::utils::clampAndWarn(impD.moment(), impDOriLower_, impDOriUpper_, "impedanceOrientationD");
+  }
+
+  /*! \brief Get the impedance spring parameter represented in the surface. */
+  const sva::ForceVecd & impedanceK() const noexcept
+  {
+    return impK_;
+  }
+
+  /*! \brief Set the impedance spring parameter represented in the surface. */
+  void impedanceK(const sva::ForceVecd & impK)
+  {
+    impK_.force() = mc_filter::utils::clampAndWarn(impK.force(), impKPosLower_, impKPosUpper_, "impedancePositionK");
+    impK_.moment() =
+        mc_filter::utils::clampAndWarn(impK.moment(), impKOriLower_, impKOriUpper_, "impedanceOrientationK");
+  }
+
+  /*! \brief Get the wrench gain, which is multiplied by the wrench in the surface frame. */
+  const sva::MotionVecd & wrenchGain() const noexcept
+  {
+    return wrenchGain_;
+  }
+
+  /*! \brief Set the wrench gain, which is multiplied by the wrench in the surface frame. */
+  void wrenchGain(const sva::MotionVecd & gain)
+  {
+    wrenchGain_.linear() =
+        mc_filter::utils::clampAndWarn(gain.linear(), wrenchGainLinLower_, wrenchGainLinUpper_, "wrenchGainLinear");
+    wrenchGain_.angular() =
+        mc_filter::utils::clampAndWarn(gain.angular(), wrenchGainAngLower_, wrenchGainAngUpper_, "wrenchGainAngular");
+  }
+
+  /*! \brief Get the desired pose of the surface in the world frame. */
+  const sva::PTransformd & desiredPose() const noexcept
```

Comment:
We use `target` rather than `desired` everywhere else so for coherency it would be better to use `target` here as well (this comment applies to every use of `desired` in this class and the associated schema)

Reply:
Originally, I intentionally used a different term than target. The desiredPose specified by the user is modified by the impedance equation, and it is passed to the target of the SurfaceTransformTask: https://github.com/mmurooka/mc_rtc/blob/519a88a24b65b0bd7472dfca4b0833aaba1d74c3/src/mc_tasks/ImpedanceTask.cpp#L141

I thought that overriding the target method in ImpedanceTask would cause the following two return values to be different, which would be confusing:
```
std::shared_ptr<ImpedanceTask> task;
task->target();
std::dynamic_pointer_cast<SurfaceTransformTask>(task)->target();
```

However, it doesn't matter in normal usage, and target may be easier for the user to understand, so either one is fine.